harness/tests/test_ipc.py

```diff
@@ -306,6 +306,32 @@ def test_worker_crashes(self) -> None:
 
             assert len(pid_server.graceful_shutdowns) == 0
 
+    def test_return_code_on_worker_error(self) -> None:
+        with ipc.PIDServer(addr=0, num_clients=2) as pid_server:
+            assert pid_server.listener
+            _, port = pid_server.listener.getsockname()
+
+            # Enforce that the crashed worker causes the exit before the other worker exits.
+            procs = [
+                multiprocessing.Process(target=TestPIDServer._worker_proc, args=(port, False, 20)),
+                multiprocessing.Process(
+                    target=TestPIDServer._worker_proc, args=(port, False, 0.5, 1, True)
+                ),
+            ]
+
+            for p in procs:
+                p.start()
+
+            error_code = pid_server.run_subprocess(
+                ["sleep", "30"],
+            )
+
+            assert error_code == 77
```

Comment:
wait, now that I read the test, I'm confused.  How is it that we are hitting the `77` codepath?  It looks to me like we should be failing with an exit code of `1` due to the `raise ValueError`, which should not return 77 in the `wait() or 77` case?

Reply:
we wait for the process given to `pid_server` which does not fail and returns 0.  however, one of the worker processes failed, which is why we still want to return a nonzero error code.

Reply:
ok, after much explaining, I now understand my own code.  Thank you.